### PR TITLE
Merge

### PR DESCRIPTION
**Pull request by nghiapham1026**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,12 +19,46 @@ async function main() {
...
+    // Fetch data from /scrape-espn endpoint
+    const response = await axios.get("http://localhost:8080/scrape-espn");
+    const headlines = response.data.headlines;
+
+    // Render the index.ejs template and pass the headlines
+    res.render("index", { headlines });
+  } catch (error) {
+    console.error("Error fetching headlines:", error);
+    res.render("index", { headlines: [] }); // Render with empty headlines
+  }
```
Comment (saravpereira) [inline index.js]: no need this block of code, app.get("/scrape-espn", ...) from below is enough

File: index.js [hunk 2]
```diff
@@ -16,12 +19,46 @@ async function main() {
...
+    $('.headlineStack__list li').each((index, element) => {
+      const headline = $(element).text().trim();
+      headlines.push(headline);
+    });
+
+    res.json({ headlines });
+  } catch (error) {
+    console.error("Error scraping:", error);
+    res.status(500).json({ error: "Error scraping data" });
+  }
```
Comment (saravpereira) [inline index.js]: move the try/catch block to a separate file under the controller folder and call it here

File: index.js (hunk 1 above)
Comment (nghiapham1026) [reply index.js]: Deleted ejs page

File: index.js (hunk 2 above)
Comment (nghiapham1026) [reply index.js]: moved to scrapeController.js

File: controllers/scrapeController.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// controller/scrapeController.js
+const axios = require("axios");
+const cheerio = require("cheerio");
+
+async function scrapeESPN(req, res) {
+  try {
+    console.log("Request received for /scrape-espn");
+    const response = await axios.get("https://www.espn.com");
```
Comment (saravpereira) [inline controllers/scrapeController.js]: move url to a new file called constants.js to store hard-coded variables that may change like urls

File: controllers/scrapeController.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// controller/scrapeController.js
+const axios = require("axios");
+const cheerio = require("cheerio");
+
+async function scrapeESPN(req, res) {
```
Comment (saravpereira) [inline controllers/scrapeController.js]: use camelCase for variable and function names names: scrapeEspn

File: controllers/scrapeController.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// controller/scrapeController.js
```
Comment (saravpereira) [inline controllers/scrapeController.js]: rename this file to webscrapeController

File: controllers/webscrapeController.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+const axios = require("axios");
+const cheerio = require("cheerio");
+
+async function scrapeEspn(req, res) {
+  try {
+    console.log("Request received for /scrape-espn");
+    const response = await axios.get("https://www.espn.com");
```
Comment (saravpereira) [inline controllers/webscrapeController.js]: https://www.espn.com needs to be in a constants.js file

**Review by saravpereira — APPROVED** [review]

**Pull request merged**